File: core-functionality/src/main/java/core/workflow/constructor/domain/entity/Module.java
package core.workflow.constructor.domain.entity;


import core.workflow.constructor.domain.entity.actions.Action;
import core.workflow.constructor.domain.entity.triggers.Trigger;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
public class Module {
    private UUID id;
    private String uniqueName;
    private List<Action> actions;
    private List<Trigger> triggers;
}

File: common-lib/src/main/kotlin/domain/Module.kt
package domain

import domain.trigger.Trigger
import java.util.UUID

data class Module(
    val id: UUID,
    val uniqueName: String,
    val api: List<Api>,
    val triggers: List<Trigger>
) {
    override fun hashCode() = id.hashCode()
    override fun equals(other: Any?): Boolean {
        if (other !is Module) return false
        return id == other.id && uniqueName == other.uniqueName
    }

}